src/model/document.ts

```diff
@@ -50,6 +50,29 @@ export class ScannedDocument {
   }
 }
 
+// A map between kind string literal types and their feature types.
+export interface FeatureKinds {
```

Comment:
I think ultimately this should be in a different file and/or and split up. PolymerElement should extend FeatureKinds with:

```typescript
declare interface FeatureKinds {
  'polymer-element': PolymerElement;
}
```
etc.